### ISSUE TITLE
Example in SC 2.5.7 seems to imply keyboard functionality acceptable if an on-screen keyboard can be used.

### ISSUE THREAD

**Issue by mraccess77** [w3c/wcag#1978]
The example in [dragging understanding](https://w3c.github.io/wcag/understanding/dragging-movements.html) about "A radial control widget where the value can be set by dragging includes a text field that presents the current value and offers the input of the desired value via a keyboard (including on-screen virtual keyboards)." seems to imply that if an on-screen keyboard can be used then a keyboard alternative for dragging is acceptable.  This seems like it negates this SC and the SC would never fail.  I'd suggest if this is not the intent that we update the example.

**Comment by patrickhlauke** [comment]
on-screen keyboards (especially those on mobile operating systems) don't generally include things like cursor keys. they also can't generally be "summoned" by the user whenever they want it, but only appear when focus is on a text entry field. so this would not negate the SC, i'd say.

however, i agree the phrasing could be changed, as it's not the keyboard bit that's important, but the fact that it doesn't need the ability to drag - so could be reworded accordingly (removing the unintended suggestion that on-screen keyboards give a pass on this in all situations)

**Comment by alastc** [comment]
I think that "can be achieved by a single pointer" would include tapping an input and using an onscreen keyboard to enter a value. That wouldn't be useful in many cases, e.g. moving an element, but should be allowed for.

So I think the example is valid. Perhaps just remove the references to keyboards?

"A radial control widget where the value can be set by dragging includes a text field that presents the current value and offers the input of the desired value via text-entry."

**Comment by mraccess77** [comment]
On-screen keyboards including those with arrow keys are available on platforms like Windows.  It can be difficult to get them to appear on mobile but that is an important point if this approach is used.  It sounds like @alastc is saying that number or text entry where a on-screen keyboard can be raised is ok - and @patrickhlauke is saying using other keys like up/down on on-screen keyboard or when on-screen keyboard can't be opened on mobile such as when a non-standard input is used then it could not be used as a technique.   So any test would need to make sure the technique worked on a mobile platform if that was needed in the conformance claim.

**Comment by detlevhfischer** [comment]
I would imagine if there are text input alternatives to something like a color wheel, these would be presented once the user activates the input. The question is whether numwerical input of a color code could ever be considered an equivalent - I guess most people will be unable to specify the color they want by inputting the exact color code (without using some other tool at least). 
As to the text input - are there cases when in spite of tapping the input the virtual keyboard would not come up? 

The easiest way (a cop-out?) would be to focus on single tap choice of color in the color wheel and add a paragraph in the understanding text to refer to the chosen a11y baseline when deciding if a virtual keyboard would equally satisfy the SC?
I can create a PR to implement that - but wait a bit for confirmation if this is the appropriate approach...

**Comment by detlevhfischer** [comment]
Following feedback in the last WCAG 2.X Issues group meeting, I have drafted PR #2009 to take out virtual keyboard as alternative.

**Comment by jake-abma** [comment]
Not sure if I follow the issue correct but an alternative is always valid in WCAG isn't it?

I do follow the fact that if dragging is implemented, there needs to be a way for single pointer without dragging.

BUT, maybe a bit late, if an alternative is present for dragging, like a numeric input field, shouldn't that be the exception for this SC?

Making the removed text valid after all?

> <li>A radial control widget where the value can be set by dragging includes a text field that presents the current value and offers the input of the desired value via a keyboard (including on-screen virtual keyboards).</li>

and also add the PR text:

> also allows picking another color value simply by tapping or clicking on another place in the color wheel.

To not negate this SC?

About the question from @detlevhfischer :

> The question is whether numerical input of a color code could ever be considered an equivalent - I guess most people will be unable to specify the color they want by inputting the exact color code (at least without using some other tool).

I do think this is an equivalent (not exactly the same but equivalent), and used a lot, specially by designers following branding colors (more than the wheel itself, not sure if they use the wheel at all) to specify the color very precise

**Comment by mraccess77** [comment]
Did we make a decision though on whether redundant input controls that require the keyboard are a way to meet this?  I understand you removed the example.

**Comment by patrickhlauke** [comment]
i would say that having a fallback/alternative that only works with keyboard would not satisfy the spirit of the SC, as the idea is that users can still operate things with a single pointer. so if, for instance, there was a keyboard event listener as an alternative, which required the user to be able to press an arbitrary key, then that would of course not work for touch/mouse users since they can't generally just summon the on-screen keyboard (on mobile/tablet operating systems anyway) and activate a key.

in the case of having text-style inputs where users can then type in things on an on-screen keyboard as an alternative, then yes that should pass as at that point it's down to the browser/operating system to bring up the on-screen keyboard, and those keyboards generally don't rely on dragging / can be operated with a single pointer.
